File: common/src/main/java/hirezapi/Configuration.java
package hirezapi;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class Configuration {
  private final Platform platform;
  private final String devId;
  private final String authKey;

  public static Configuration of(final Platform platform, final String devId, final String authKey) {
    return new Configuration(platform, devId, authKey);
  }
}

File: api/src/main/kotlin/hirez/Configuration.kt
package hirez

import hirez.enums.Language

/**
 *
 * @author Damian Staszewski [damian@stachuofficial.tv]
 * @version %I%, %G%
 * @since 1.0
 */
data class Configuration(
			var baseEndpoint: BaseEndpoint,
			val devId: String,
			val authKey: String,
			var defaultLanguage: Language = Language.English
) {
	constructor(
				baseEndpoint: BaseEndpoint,
				devId: String,
				authKey: String
	) : this(baseEndpoint, devId, authKey, Language.English)
}